qiling/os/windows/dlls/kernel32/fileapi.py

```diff
@@ -269,3 +269,50 @@ def hook_GetShortPathNameW(ql, address, params):
     else:
         ql.mem.write(dst, res)
     return len(res) - 1
+
+
+# BOOL GetVolumeInformationW(
+#   LPCWSTR lpRootPathName,
+#   LPWSTR  lpVolumeNameBuffer,
+#   DWORD   nVolumeNameSize,
+#   LPDWORD lpVolumeSerialNumber,
+#   LPDWORD lpMaximumComponentLength,
+#   LPDWORD lpFileSystemFlags,
+#   LPWSTR  lpFileSystemNameBuffer,
+#   DWORD   nFileSystemNameSize
+# );
+@winapi(cc=STDCALL, params={
+    "lpRootPathName": POINTER,
+    "lpVolumeNameBuffer": POINTER,
+    "nVolumeNameSize": DWORD,
+    "lpVolumeSerialNumber": POINTER,
+    "lpMaximumComponentLength": POINTER,
+    "lpFileSystemFlags": POINTER,
+    "lpFileSystemNameBuffer": POINTER,
+    "nFileSystemNameSize": DWORD
+})
+def hook_GetVolumeInformationW(ql, address, params):
+    root_pt = params["lpRootPathName"]
+    if root_pt != 0:
+        root = read_wstring(ql, root_pt)
+        pt_volume_name = params["lpVolumeNameBuffer"]
+        if pt_volume_name != 0:
+            # TODO implement
+            volume_name = ("AAAABBBB"+"\x00").encode("utf-16le")
+            ql.uc.mem_write(pt_volume_name, volume_name)
```

Comment:
all the ql.uc.mem_write please change to ql.mem.write same goes to ql.mem.read and ql.mem.map
referr to qiling/os/memory.py

Reply:
Sure, forgot about it!